### PR TITLE
support internlm2-reward

### PR DESCRIPTION
**Pull request by RangiLyu**
## Motivation

Support InternLM2-Reward https://huggingface.co/collections/internlm/internlm2-reward-669a0b17cab8a9d2c3016601

## Modifications

- add internlm2_reward.py in python/sglang/srt/models.
- edit supported reward model in docs/references/supported_models.md.
- edit python/sglang/srt/configs/model_config.py

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/models/internlm2_reward.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import torch
+from torch import nn
+from transformers import PretrainedConfig
+
+from sglang.srt.layers.pooler import EmbeddingPoolerOutput, Pooler, PoolingType
+from sglang.srt.layers.quantization.base_config import QuantizationConfig
+from sglang.srt.model_executor.forward_batch_info import ForwardBatch
+
+from .internlm2 import InternLM2ForCausalLM, InternLM2Model
```
Comment (merrymercy) [inline python/sglang/srt/models/internlm2_reward.py]:
```suggestion
from sglang.srt.models.internlm2 import InternLM2ForCausalLM, InternLM2Model
```

File: python/sglang/srt/models/internlm2_reward.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import torch
+from torch import nn
+from transformers import PretrainedConfig
+
+from sglang.srt.layers.pooler import EmbeddingPoolerOutput, Pooler, PoolingType
+from sglang.srt.layers.quantization.base_config import QuantizationConfig
+from sglang.srt.model_executor.forward_batch_info import ForwardBatch
+
+from sglang.srt.models.internlm2 import InternLM2ForCausalLM, InternLM2Model
```
Comment (merrymercy) [inline python/sglang/srt/models/internlm2_reward.py]:
```suggestion
from sglang.srt.models.internlm2 import InternLM2ForCausalLM, InternLM2Model
```
